### ISSUE TITLE
no-unused-variable false positive when variable is used in JSX spread

### ISSUE THREAD

**Issue by myitcv** [palantir/tslint#658]
Re-created from #570, specifically [this comment](https://github.com/palantir/tslint/issues/570#issuecomment-138332733) which wasn't addressed by #655 

```typescript
/// <reference path="tsd.d.ts" />

import * as React from "react";
interface IMProps {
	Name: string;
}

export class HelloMsg extends React.Component<{}, {}> {
	constructor() {
		super();
	}
	render(): JSX.Element {
		let props: IMProps = { Name: "test" };

		return <div>
				<Msg {...props} />
			</div>;
	}
}

export class Msg extends React.Component<IMProps, {}> {
	constructor() {
		super();
	}
	render(): JSX.Element {
		return <div></div>;
	}
}
```

```
$ tslint -c app/tslint.json app/**/*.{ts,tsx}
app/example.tsx[13, 7]: unused variable: 'props'
```

Configuration as follows:

* TypeScript `v1.6.0-dev.20150907`
* `tslint v2.5.0-dev.5`
* `tsconfig.json`:

```
{
  "compilerOptions": {
    "target": "es5",
    "module": "system",
    "declaration": false,
    "noImplicitAny": true,
    "removeComments": false,
    "noLib": false,
    "sourceMap": true,
    "jsx": "react",
    "rootDir": ".",
    "outDir": "../dist/"
  }
}
```

* [`tslint.json`](https://gist.github.com/myitcv/783b55f3e65720de2072)

The underlying problem appears to be a TypeScript core issue, tracked by [this issue](https://github.com/Microsoft/TypeScript/issues/4688)

**Comment by adidahiya** [comment]
update: the compiler bug is being fixed here https://github.com/Microsoft/TypeScript/pull/5040

**Comment by RyanCavanaugh** [comment]
Microsoft/TypeScript#5040 has been fixed in the TypeScript `master` branch.

**Comment by myitcv** [comment]
@adidahiya - please can we get a new `next` version cut?

**Comment by myitcv** [comment]
Thanks @adidahiya - confirm this is fixed as of `v2.6.0-dev.2`
